lib/OpenQA/Worker/Common.pm

```diff
@@ -432,29 +450,12 @@ sub register_worker {
         }
         return;
     }
-    my $newid    = $tx->success->json->{id};
-    my $workerid = $hosts->{$host}{workerid};
-    my $ws       = $hosts->{$host}{ws};
-    if ($ws && $workerid && $workerid != $newid) {
-        # terminate websockets if our worker id changed
-        $ws->finish() if $ws;
-        $ws = undef;
-    }
+    my $newid = $tx->success->json->{id};
```

Comment:
The `success` method is rarely used anymore (only in `if (...) {` conditions). This looks like it should be `$tx->result` or `$tx->res`, depending on if you want to die on connection errors.